src/unstable/MessageActions/MessageActions.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable sort-keys */
+import clsx from 'clsx';
+import React, { PropsWithChildren, useState } from 'react';
+
+import { useChatContext, useMessageContext, useTranslationContext } from '../../context';
+import { ActionsIcon } from '../../components/Message/icons';
+import { DialogAnchor, useDialog, useDialogIsOpen } from '../../components/Dialog';
+import { MessageActionsWrapper } from '../../components/MessageActions/MessageActions';
+
+import { useBaseMessageActionSetFilter, useSplitMessageActionSet } from './hooks';
+import { defaultMessageActionSet } from './defaults';
+
+export type MessageActionSetItem = {
+  Component: React.ComponentType;
```

Comment:
Not sure if there's a good way to enforce non-null return type - any suggestions are welcome.